packages/dev/core/src/Cameras/camera.ts

```diff
@@ -1572,4 +1572,31 @@ export class Camera extends Node {
 
         return handednessMultiplier;
     }
+    /**
+     * Query a hotspot
+     * @param hotSpotQuery info necessary to compute the hotspot
+     * @param res resulting hotspot infor
+     * @returns true if hotspot could be computed
+     */
+    getHotSpotToRef(hotSpotQuery: HotSpotQuery, res: HotSpot): boolean {
+        const scene = this.getScene();
+        if (hotSpotQuery.meshIndex >= scene.meshes.length) {
+            return false;
+        }
+        const mesh = scene.meshes[hotSpotQuery.meshIndex];

```

Comment:
Why using a `meshIndex` instead of a `mesh` in `HotSpotQuery`? That would get rid of the scene dependency. Also, if meshes are added or removed to the scene, it could invalidate the index.

Reply:
if meshes are added or removed, user can use `getHotSpotToRef` from `abstractMesh`. That would happen in a more complex situation than the Viewer. This camera method is intended to only work with the viewer. Maybe it should move in that class actually.

Reply:
I understand, but is there an advantage to using an index rather than a mesh? I think it's easier for people to provide a mesh, because that's usually what they have. To get an index, you'd have to do `scene.meshes.indexOf(myMesh)`.

Reply:
In the context of the viewer, inputs are a bit different. Maybe providing the mesh name instead of its index makes more sense in the data-surface. something like `data-surface="myMesh 0 228 113 111 0.217 0.341 0.442"` ? @ryantrem

Reply:
If the viewer takes an index, you could simply do `scene.meshes[index]` yourself before calling the hotspot function? Or allow both an index and a mesh in the input structure: if the mesh is provided, use it, otherwise use the index to lookup the mesh in `scene.meshes`.